views/index.hbs

```diff
@@ -1,2 +1,109 @@
-<h1>{{title}}</h1>
-<p>Welcome to {{title}}</p>
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta name="viewport" content="initial-scale=1.0, user-scalable=no">
+    <meta charset="utf-8">
+    <title>Marker Clustering</title>
+    <style>
+      /* Always set the map height explicitly to define the size of the div
+       * element that contains the map. */
+      #map {
+        height: 100%;
+      }
+      /* Optional: Makes the sample page fill the window. */
+      html, body {
+        height: 100%;
+        margin: 0;
+        padding: 0;
+      }
+    </style>
+    <script type='text/javascript' src='https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js'></script>
+  </head>
+  <body>
+    <div id="map"></div>
+    <script>
+
+      function initMap() {
+
+        var map = new google.maps.Map(document.getElementById('map'), {
+          zoom: 3,
+          center: {lat: -28.024, lng: 140.887}
+        });
+
+        // Create an array of alphabetical characters used to label the markers.
+        var labels = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+
+        // Add some markers to the map.
+        // Note: The code uses the JavaScript Array.prototype.map() method to
+        // create an array of markers based on a given "locations" array.
+        // The map() method here has nothing to do with the Google Maps API.
+        var markers = bikeLocations.map(function(location, i) {
+          return new google.maps.Marker({
+            position: location,
+            label: labels[i % labels.length]
+          });
+        });
+
+        // Add a marker clusterer to manage the markers.
+        var markerCluster = new MarkerClusterer(map, markers,
+            {imagePath: 'https://developers.google.com/maps/documentation/javascript/examples/markerclusterer/m'});
+      }
+      var networks = [];
+	  $.ajax({
+	      dataType: 'json',
+	      data: $('#formID').serialize(),
+	      type: 'GET',
+	      url: "http://localhost:3001/networks",
+	      success: function (data) {
+	        // console.log('success', data);
+	        for (var i in data) {
+			  var item = data[i];
+			  // console.log(item['latitude']);
+			  // console.log(item['longitude']);
+			  networks.push(item["_id"]);
+
+			}
+	      },
+	      async: false,
+	      error: function (XMLHttpRequest, textStatus, errorThrown) {
+	        console.log('error', errorThrown);
+	      }
+	    });
+	  // console.log(networks);
+
+
+
+      var bikeLocations = []
+      // 5993a2be0d43ff1e8c7f72e0 doesnt show any markers wtf? --> google maps shows this should appear abu dhabi
+      for (var i in networks){
```

Comment:
I am not sure what is going on here. It seems to me you are trying to get locations for all the networks for every map load. This will easily overload your server.

The requirement was to actually get the locations for the map on click on a network. This means that when you click on a network, you will get the locations of all it's stations. 

This seems that you are getting all bike locations for all networks at the same time.

How many of these do you have in the database?